ui/selectors/approvals.ts

```diff
@@ -0,0 +1,19 @@
+import { ApprovalRequest } from '@metamask/approval-controller';
+import { ApprovalType, Json } from '@metamask/controller-utils';
+
+type ApprovalsMetaMaskState = {
```

Comment:
This is a minimal type for the metamask state that represents what we need for this selector. There is no type that fully defines the state.